spec/std/float_printer/ryu_printf_spec.cr

```diff
@@ -0,0 +1,3187 @@
+  {1e160, 0, 160},
+  {1e161, 0, 161},
+  {1e162, 0, 161},
+  {1e163, 0, 161},
+  {1e164, 0, 164},
+  {1e165, 0, 164},
+  {1e166, 0, 165},
+  {1e167, 0, 167},
+  {1e168, 0, 167},
+  {1e169, 0, 167},
+  {1e170, 0, 170},
+  {1e171, 0, 170},
+  {1e172, 0, 172},
+  {1e173, 0, 173},
+  {1e174, 0, 174},
+  {1e175, 0, 174},
+  {1e176, 0, 176},
+  {1e177, 0, 176},
+  {1e178, 0, 178},
+  {1e179, 0, 178},
+  {1e180, 0, 180},
+  {1e181, 0, 180},
+  {1e182, 0, 182},
+  {1e183, 0, 182},
+  {1e184, 0, 184},
+  {1e185, 0, 184},
+  {1e186, 0, 184},
+  {1e187, 0, 186},
+  {1e188, 0, 188},
+  {1e189, 0, 188},
+  {1e190, 0, 190},
+  {1e191, 0, 190},
+  {1e192, 0, 192},
+  {1e193, 0, 193},
+  {1e194, 0, 193},
+  {1e195, 0, 194},
+  {1e196, 0, 195},
+  {1e197, 0, 195},
+  {1e198, 0, 198},
+  {1e199, 0, 199},
+  {1e200, 0, 199},
+  {1e201, 0, 201},
+  {1e202, 0, 200},
+  {1e203, 0, 202},
+  {1e204, 0, 202},
+  {1e205, 0, 205},
+  {1e206, 0, 206},
+  {1e207, 0, 206},
+  {1e208, 0, 207},
+  {1e209, 0, 209},
+  {1e210, 0, 209},
+  {1e211, 0, 210},
+  {1e212, 0, 211},
+  {1e213, 0, 212},
+  {1e214, 0, 213},
+  {1e215, 0, 214},
+  {1e216, 0, 216},
+  {1e217, 0, 216},
+  {1e218, 0, 217},
+  {1e219, 0, 218},
+  {1e220, 0, 219},
+  {1e221, 0, 221},
+  {1e222, 0, 221},
+  {1e223, 0, 221},
+  {1e224, 0, 223},
+  {1e225, 0, 224},
+  {1e226, 0, 225},
+  {1e227, 0, 225},
+  {1e228, 0, 227},
+  {1e229, 0, 228},
+  {1e230, 0, 230},
+  {1e231, 0, 231},
+  {1e232, 0, 231},
+  {1e233, 0, 232},
+  {1e234, 0, 234},
+  {1e235, 0, 235},
+  {1e236, 0, 235},
+  {1e237, 0, 235},
+  {1e238, 0, 238},
+  {1e239, 0, 238},
+  {1e240, 0, 240},
+  {1e241, 0, 241},
+  {1e242, 0, 241},
+  {1e243, 0, 243},
+  {1e244, 0, 243},
+  {1e245, 0, 245},
+  {1e246, 0, 246},
+  {1e247, 0, 246},
+  {1e248, 0, 248},
+  {1e249, 0, 247},
+  {1e250, 0, 247},
+  {1e251, 0, 251},
+  {1e252, 0, 252},
+  {1e253, 0, 252},
+  {1e254, 0, 252},
+  {1e255, 0, 254},
+  {1e256, 0, 256},
+  {1e257, 0, 256},
+  {1e258, 0, 258},
+  {1e259, 0, 258},
+  {1e260, 0, 260},
+  {1e261, 0, 258},
+  {1e262, 0, 262},
+  {1e263, 0, 262},
+  {1e264, 0, 264},
+  {1e265, 0, 265},
+  {1e266, 0, 266},
+  {1e267, 0, 266},
+  {1e268, 0, 266},
+  {1e269, 0, 269},
+  {1e270, 0, 269},
+  {1e271, 0, 270},
+  {1e272, 0, 272},
+  {1e273, 0, 272},
+  {1e274, 0, 273},
+  {1e275, 0, 274},
+  {1e276, 0, 276},
+  {1e277, 0, 277},
+  {1e278, 0, 277},
+  {1e279, 0, 279},
+  {1e280, 0, 280},
+  {1e281, 0, 280},
+  {1e282, 0, 280},
+  {1e283, 0, 282},
+  {1e284, 0, 284},
+  {1e285, 0, 284},
+  {1e286, 0, 286},
+  {1e287, 0, 287},
+  {1e288, 0, 288},
+  {1e289, 0, 289},
+  {1e290, 0, 289},
+  {1e291, 0, 290},
+  {1e292, 0, 292},
+  {1e293, 0, 292},
+  {1e294, 0, 294},
+  {1e295, 0, 294},
+  {1e296, 0, 294},
+  {1e297, 0, 297},
+  {1e298, 0, 297},
+  {1e299, 0, 299},
+  {1e300, 0, 299},
+  {1e301, 0, 299},
+  {1e302, 0, 302},
+  {1e303, 0, 303},
+  {1e304, 0, 303},
+  {1e305, 0, 303},
+  {1e306, 0, 306},
+  {1e307, 0, 306},
+  {1e308, 0, 308},
+]
```

Comment:
**suggestion:** We can prevent materializing this array by breaking the typing. This will work find from macro land, but errors when the constant is referenced from regular code.
```suggestion
] of _
```